ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -169,4 +175,21 @@ internal constructor(
   }
 
   fun getZoomThresholdCrossed(): Observable<Nil> = zoomThresholdCrossed
+
+  private fun getLocationOfInterestFeatures(survey: Survey): Flow<Set<Feature>> =
+    loiRepository.getLocationsOfInterest(survey).map {
+      it.map { loi -> loi.toFeature() }.toPersistentSet()
+    }
+
+  private suspend fun LocationOfInterest.toFeature() =
+    Feature(
+      id = id,
+      type = FeatureType.LOCATION_OF_INTEREST.ordinal,
+      flag =
+        submissionCount + submissionRepository.getPendingCreateCount(id) -
+          submissionRepository.getPendingDeleteCount(id) > 0,
```

Comment:
Can we move these to separate lines for better readability?

Reply:
Sure, coming right up